tests/framework/installer/test_data.go

```diff
@@ -0,0 +1,93 @@
+package installer
+
+import (
+	"github.com/rook/rook/tests/framework/utils"
+)
+
+func GetBlockPoolDef(poolName string, namespace string, replicaSize string) string {
+	return `apiVersion: rook.io/v1alpha1
+kind: Pool
+metadata:
+  name: ` + poolName + `
+  namespace: ` + namespace + `
+spec:
+  replicated:
+    size: ` + replicaSize
+}
+func BlockPoolOperation(k8sh *utils.K8sHelper, poolName string, namespace string, replicaSize string, action string) (string, error) {
+	poolStorageClassDef := GetBlockPoolDef(poolName, namespace, replicaSize)
+	return k8sh.ResourceOperation(action, poolStorageClassDef)
+}
+
+func GetBlockStorageClassDef(poolName string, storageClassName string, namespace string) string {
+	return `apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: ` + storageClassName + `
+provisioner: rook.io/block
+parameters:
+    pool: ` + poolName + `
+    clusterName: ` + namespace
+}
+
+func BlockStorageClassOperation(k8sh *utils.K8sHelper, poolName string, storageClassName string, namespace string, action string) (string, error) {
+	poolStorageClassDef := GetBlockStorageClassDef(poolName, storageClassName, namespace)
+	return k8sh.ResourceOperation(action, poolStorageClassDef)
+}
+
+func BlockPvcDef(claimName string, storageClassName string) string {
+	return `apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: ` + claimName + `
+  annotations:
+    volume.beta.kubernetes.io/storage-class: ` + storageClassName + `
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1M`
+}
+
+func BlockPvcOperation(k8sh *utils.K8sHelper, claimName string, storageClassName string, action string) (string, error) {
+	pvcDef := BlockPvcDef(claimName, storageClassName)
+	return k8sh.ResourceOperation(action, pvcDef)
+}
+
+func GetBlockPoolStorageClassAndPvcDefinition(k8sh *utils.K8sHelper, namespace string, poolName string, storageclassName string, blockName string) string {
+	return `apiVersion: rook.io/v1alpha1
```

Comment:
long term we are likely going to have other places in the tests that want to concatenate yaml snippets together. I'm suggesting that you still don't need to duplicate this. This could be implemented as follows:
```
func concatYaml(first, second string) string {
  return first +`
---
` + second
}
func GetBlockPoolStorageClassAndPvcDefinition(<args>) string {
  return concatYaml(GetBlockPoolDef(<args>), concatYaml(GetBlockStorageClassDef(<args>), BlockPvcDef(<args>))
}
```

Reply:
That is a good Idea, thanks. I did another refactor .